### DISCUSSION TITLE
Provide an autobuild for a Jamulus with JACK windows version.

### DISCUSSION THREAD

**Discussion by jujudusud** [jamulussoftware/jamulus#1750] [General]
I was thinking about the possibility to use JACK server on windows by building your own custom "jackonwindows" Jamulus.
I think we should provide a package (.exe installer) for windows of a version of Jamulus using JACK server on this OS since nearly 99% of Windows users just want to install software instead of build there own packages.

**Comment by henkdegroot** [comment]
I agree, most Windows users won't go and build their own packages.
Should this exe installer have a different name compared to the current installer?
Should there be a visual aspect in the app that is a "JACK" version instead of an "ASIO" version?

**Reply by jujudusud** [reply]
I think we can provide a package with something like `for JACK` or `Jackified` additionnal text in the name since it's a really différent version, for now non compatible with the ASIO one.
I should to be a replacement of the ASIO version and just stands in the same place from my point of view.

**Reply by ann0see** [reply]
Yes, JACK on Windows would be good; however it will not move me to use Windows for Audio stuff. It’s just not as well integrated as on linux. However, some ideas:

1. You can probably modify the ps script to generate two eye files
2. You can modify the installer with an option to skip the ASIO check
@npostavs already added a separate build for the PortAudio PR

**Reply by jujudusud** [reply]
@ann0see, Do we provide a PR for that with its own package ?

**Reply by ann0see** [reply]
See https://github.com/jamulussoftware/jamulus/pull/870

**Reply by jujudusud** [reply]
I have read that but it is not exactly the same thing.
Jack support is implemented and actually worked well.
The only missing thing is an autobuild package.

@henkdegroot do you please PR your branch ?

**Reply by henkdegroot** [reply]
@jujudusud I can create a PR for this, no problem (already had autobuild for this working in my repo, so I know what to do).
However my PR depends on PR #1718 being merged first, so I can work on it but can't complete it.

There is an automated build installer version available in my repo here: [https://github.com/henkdegroot/jamulus/releases/tag/3.8.0rc1dev](https://github.com/henkdegroot/jamulus/releases/tag/3.8.0rc1dev)

**Reply by jujudusud** [reply]
I hope #1718 will be merged soon.

**Reply by hoffie** [reply]
Having autobuild packages would be a good thing, yes.

>  Should this exe installer have a different name compared to the current installer?

Yes, definitely. All of the Windows documentation is centered around the regular, ASIO-based version and we should keep it like that unless the jack version starts being easier to setup or something. Therefore, I consider it very important to distinguish both builds and mark the jack one as "non-normal" for now.

>  Should there be a visual aspect in the app that is a "JACK" version instead of an "ASIO" version?

That's a good question. It may be important for other build flags as well. So maybe we should split this task off this issue into a more generic "how to define a Jamulus build"?

One option I could think of would be including the value of `CONFIG=` into the binary (maybe even build-time Qt version, compiler versions, if possible). This could be shown in the help screen or in `Jamulus -v`. That way it would not be disturbing to users, but we would still have a specific thing to ask for when gathering information in bug reports.

**Reply by henkdegroot** [reply]
> So maybe we should split this task off this issue into a more generic "how to define a Jamulus build"?

Sound like a good plan to do a more generic one and add info to help screen or the -v output.

**Reply by henkdegroot** [reply]
> 2\. You can modify the installer with an option to skip the ASIO check

@ann0see , should the installer with jack be as clever as the ASIO version and check if JACK is installed or not?
And if not, provide similar dialogs as the ASIO version?
Or can we state that people who choose to use JACK should know they have to install JACK as well?

**Reply by ann0see** [reply]
> should the installer with jack be as clever as the ASIO version and check if JACK is installed or not?

Definitly not a requirement, but would be nice. You could check if jack.exe exists and show a similar warning.

**Reply by henkdegroot** [reply]
> You could check if jack.exe exists and show a similar warning.

ASIO check is based on a registry key, we can do the same thing for JACK. It does mean we need to add strings to the installer nsi files for all languages.

**Reply by henkdegroot** [reply]
> It does mean we need to add strings to the installer nsi files for all languages.

To keep the introduction simple, I will setup my PR without this. It can always be added after to improve.

**Reply by henkdegroot** [reply]
@ann0see ,  I have a working solution for the scripts etc. The complete change will add deploy scripts for windows autobuild and a jack powershell script as well. To make the autobuild include this, the overall autobuild steps have to be updated as well. Can that all be done in 1 PR or should the "autobuild" update be a separate PR after the initial one has been merged?

**Reply by ann0see** [reply]
I‘d use one PR to keep it simple
